include/vara/Solver/Solver.h

```diff
@@ -131,6 +131,16 @@ class Solver {
   /// unsatisfiable).
   virtual Result<SolverErrorCode, std::unique_ptr<vara::feature::Configuration>>
   getNextConfiguration() = 0;
+
+  // TODO (@boehmseb): Was this initial that there is not a decl. in the class?
+  // ++ This is not part of the Solver interface, which seems also strange
+  // (wrong)
+
+  virtual Result<SolverErrorCode,
+                 std::vector<std::unique_ptr<vara::feature::Configuration>>>
+  getAllValidConfigurations() = 0;
+
+  virtual Result<SolverErrorCode, uint64_t> getNumberValidConfigurations() = 0;
```

Comment:
This part was removed by 864ff022ee and should stay removed.
The problem here is that these functions modify solver state in a way that interferes with the result of other solver functions.

I think this PR merged an older version of the branch that eventually was merged as 864ff022ee and therefore contains some outdated code. I try to mark all the relevant locations, but as far as I can see, all changes to the solver code in this PR should be reverted.

```suggestion
```